src/devices/purifier/class.ts

```diff
@@ -0,0 +1,217 @@
+import { Device } from '../../device';
+import {
+	BinaryPropertyStatus,
+	PropertyStatus,
+	StatusCode,
+	valueBinary,
+	ValueSingle,
+	ValueType,
+	mode,
+} from './properties';
+
+export class Purifier extends Device {
+	constructor(props) {
+		super(props);
+	}
+
+	/**
+	 * Queues a power on action
+	 */
+	queuePowerOn(): void {
+		this.queuePropertyStatusUpdate({
+			valueSingle: {
+				code: ValueSingle.POWER_ON,
+			},
+			statusCode: StatusCode.POWER,
+			valueType: ValueType.SINGLE,
+		});
+	}
+
+	/**
+	 * Queues a power off action
+	 */
+	queuePowerOff(): void {
+		this.queuePropertyStatusUpdate({
+			valueSingle: {
+				code: ValueSingle.POWER_OFF,
+			},
+			statusCode: StatusCode.POWER,
+			valueType: ValueType.SINGLE,
+		});
+	}
+
+	/**
+	 * Convert mode string to binary.
+	 * @return     {string}  mode binary
+	 */
+	existMode = (mode: string): string => {
+		switch (mode) {
+			case 'pollen':
+				return valueBinary.OPERATION_POLLEN;
+			case 'realization':
+				return valueBinary.OPERATION_REALIZE;
+			case 'ai_auto':
+				return valueBinary.OPERATION_AI_AUTO;
+			case 'auto':
+				return valueBinary.OPERATION_AUTO;
+			case 'night':
+				return valueBinary.OPERATION_NIGHT;
+			case 'silent':
+				return valueBinary.OPERATION_SILENT;
+			case 'medium':
+				return valueBinary.OPERATION_MEDIUM;
+			case 'high':
+				return valueBinary.OPERATION_HIGH;
+			default:
+				return '';
+		}
+	};
+
+	/**
+	 * Queues a specific propertyStatus for change
+	 * This alone does not do anything, the changes need to get transmitted to the
+	 * cocoro air api
+	 *
+	 * @param      {PropertyStatus}  propertyStatus  The property status
+	 */
+	queuePropertyStatusUpdate(propertyStatus: PropertyStatus): void {
```

Comment:
This looks identical to device.ts, so we can remove it here. Use the `super.queuePropertyStatusUpdate`

Reply:
There is a reason for this, the type PropertyStatus is slightly different

I didn't understand the extend part of the type, and I thought I would have to change the device.ts property file to make it the same as device.ts, so I temporarily wrote it separately.

I thought that OPERATION_MODE was unique and included only in the air purifier, so I wanted to separate it, is there any way to solve this?

```typescript
export enum StatusCode {
  STATE_DETAIL = 'FA',
  WINDSPEED = 'A0',
  POWER = '80',
  NANOE_MODE = 'CF',
  OPERATION_MODE = 'B0',
  HEAT_TEMPERATURE = 'B3',
  ROOM_TEMPERATURE = 'BB',
}
```

↓

```typescript
export enum StatusCode {
  STATE_DETAIL = 'FA',
  WINDSPEED = 'A0',
  POWER = '80',
  NANOE_MODE = 'CF',
  OPERATION_MODE = 'B0',
  HEAT_TEMPERATURE = 'B3',
  ROOM_TEMPERATURE = 'BB',
  OPERATION_MODE = 'F3', // ← added this
}
```

Reply:
I'm sorry, but the solution to this was to make it super and make it as common as possible.
I've made it any, and I know that's not a good solution, so best practices, please!